sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClientBuilder.java

```diff
@@ -0,0 +1,144 @@
+     *
+     * @param networkInterceptor the interceptor to add
+     * @return the builder
+     */
+    public OkHttpAsyncHttpClientBuilder networkInterceptor(Interceptor networkInterceptor) {
+        Objects.requireNonNull(networkInterceptor);
+        this.httpClientBuilder.addNetworkInterceptor(networkInterceptor);
+        return this;
+    }
+
+    /**
+     * Set the read timeout, default is 120 seconds.
+     *
+     * @param timeout the timeout
+     * @param unit the time unit for the timeout
+     * @return the builder
+     */
+    public OkHttpAsyncHttpClientBuilder readTimeout(long timeout, TimeUnit unit) {
+        this.httpClientBuilder.readTimeout(timeout, unit);
+        return this;
+    }
+
+    /**
+     * Set the connection timeout, default is 60 seconds.
+     *
+     * @param timeout the timeout
+     * @param unit the time unit for the timeout
+     * @return the builder
+     */
+    public OkHttpAsyncHttpClientBuilder connectionTimeout(long timeout, TimeUnit unit) {
+        this.httpClientBuilder.connectTimeout(timeout, unit);
+        return this;
+    }
+
+    /**
+     * Sets the Http connection pool.
+     *
+     * @param connectionPool the OkHttp connection pool to use
+     * @return the builder
+     */
+    public OkHttpAsyncHttpClientBuilder connectionPool(ConnectionPool connectionPool) {
+        this.connectionPool = Objects.requireNonNull(connectionPool, "connectionPool == null");
+        return this;
+    }
+
+    /**
+     * Sets the dispatcher that also composes  the thread pool for executing HTTP requests.
+     *
+     * @param dispatcher the dispatcher to use
+     * @return the builder
+     */
+    public OkHttpAsyncHttpClientBuilder dispatcher(Dispatcher dispatcher) {
+        this.dispatcher = Objects.requireNonNull(dispatcher, "dispatcher == null");
+        return this;
+    }
+
+    /**
+     * Sets the proxy.
+     *
+     * @param proxy the proxy
+     * @return the builder
+     */
+    public OkHttpAsyncHttpClientBuilder proxy(java.net.Proxy proxy) {
+        this.httpClientBuilder.proxy(proxy);
+        return this;
+    }
+
+    /**
+     * Sets the proxy authenticator.
+     *
+     * @param proxyAuthenticator the proxy authenticator
+     * @return the builder
+     */
+    public OkHttpAsyncHttpClientBuilder proxyAuthenticator(Authenticator proxyAuthenticator) {
+        this.httpClientBuilder.proxyAuthenticator(proxyAuthenticator);
+        return this;
+    }
+
+    /**
+     * Build a HttpClient with current configurations.
+     *
+     * @return a {@link HttpClient}.
+     */
+    public HttpClient build() {
+        if (connectionPool != null) {
+            this.httpClientBuilder = httpClientBuilder.connectionPool(connectionPool);
+        }
+        if (dispatcher != null) {
+            this.httpClientBuilder = httpClientBuilder.dispatcher(dispatcher);
+        }
+        return new OkHttpAsyncHttpClient(this.httpClientBuilder.build());
+    }
```

Comment:
I think this builder pattern needs consideration. For example, it would not be possible for person to create two HttpClient instances from this builder where the first one has a dispatcher and the second one does not, because they are unable to null it out. I would probably remove the constructor that takes the HttpClientBuilder and to only create that in the build() method here.

Reply:
Yes, we should support that